tripleblindmarket/private-kit
# Global Store Introduction

#### Description:

Adds a global redux store to the app, with persistence & an action / reducer for replacing the on boarding async storage flow. 

The motivation for this is to set up the infrastructure for sharing of cross-component state in a way that is more easily consumable by different screens & components, ensures a shared, normalized data layer for our upcoming APIs, and allows us to more gracefully handle (and ommit) loading states.

Additionally, we can debug our coupled state better (I've set up RN debugger integrations for Redux Dev Tools here). 

Replacing the onboarding async store with a persited redux state allows us to have a state of "onboardingComplete" at mount time for the navigator, stopping navigator flash between onboarding and main. Follow up is needed to keep the splash screen up during the persist gate loading.

#### Screenshots:
Behavior of the app load with the persist gate (note the button text flash on the before):

Before             |  After
:-------------------------:|:-------------------------:
![2020-05-27 22 17 37](https://user-images.githubusercontent.com/25315679/83090953-00484280-a068-11ea-8b15-4ab028bfd30c.gif) | ![2020-05-27 22 20 04](https://user-images.githubusercontent.com/25315679/83091121-78af0380-a068-11ea-86bc-2a917e430d7e.gif)







#### How to test:
Ensure that going through the onboarding flow can only occur once. 
No app functionality should change.
Can additionally now use Redux Dev Tools / RN Debugger to ensure that redux state is persisted / updated properly after onboarding flow.
